beacon-chain/cache/depositsnapshot/deposit_tree.go

```diff
@@ -100,10 +100,19 @@ func (d *DepositTree) getProof(index uint64) ([32]byte, [][32]byte, error) {
 		return [32]byte{}, nil, ErrInvalidDepositCount
 	}
 	finalizedDeposits, _ := d.tree.GetFinalized([][32]byte{})
+	finalizedIdx := -1
 	if finalizedDeposits != 0 {
-		finalizedDeposits = finalizedDeposits - 1
+		fd, err := math.Int(finalizedDeposits)
+		if err != nil {
+			return [32]byte{}, nil, err
+		}
+		finalizedIdx = fd - 1
 	}
-	if index <= finalizedDeposits {
+	i, err := math.Int(index)
+	if err != nil {
+		return [32]byte{}, nil, err
+	}
+	if finalizedDeposits > 0 && i <= finalizedIdx {
```

Comment:
Not exactly related to the issue that we resolved but if you provide an invalid index(non-existent in tree) as an argument to this method, this particular method will panic. We should also be asserting that any provided index is less than the deposit count.